review/searchTab.js

```diff
@@ -0,0 +1,108 @@
+import { CLASS_LIST, URL, KEYCODE } from '../util/data';
+import { li, makeAutoCompleteItem, makeRecommendItem, ul } from '../util/htmlTemplate';
+import { autoCompleteParser } from '../util/parser';
+import { getData, _, debounce } from '../util/util';
+
+function SearchTab({ data, selector }) {
+  this.recommendData = data;
+  this.searchTab = selector.searchTab;
+  this.searchInput = selector.searchInput;
+  this.searchTabTitle = selector.searchTabTitle;
+  this.currentIdx = -1;
+  this.orginInput;
+  this.autoCompleteData;
+  this.timer;
+}
+
+SearchTab.prototype = {
+  constructor: SearchTab,
+  init() {
+    this.registerEvent();
+    this.renderSearchTab();
+  },
+  registerEvent() {
+    this.searchInput.addEventListener('input', this.handleInput.bind(this));
+    this.searchInput.addEventListener('keydown', this.handleKeydown.bind(this));
+  },
+  handleInput({ target: { value } }) {
+    this.timer = debounce(this.render.bind(this, value), 1000, this.timer);
```

Comment:
debounce의 개념은 잘 이해하고 구현해 주신 것 같아요 👍  다만 일반적으로 debounce함수는 함수를 반환하도록 구현되어 있고 현재 구현은 timer를 debounce의 인자로 넘겨주는 점이 조금 아쉽네요. debounce를 사용하는 곳에서 timer를 다루지 않아도 되도록 lodash의 debounce함수 예제를 참고하셔서 (https://lodash.com/docs/#debounce) 아래와 같이 변경해보면 좋을 것 같아요.
```suggestion
handleInput: debounce(
  function ({ target: { value } }) {
    this.render.(value);
  }, 1000),
```

Reply:
링크까지 공유해주셔서 감사합니다! lodash 디바운스 코드가 이해가 쉽지 않지만 꼭 이해하고 활용도 해보도록 하겠습니다!